1.30.2

matteobruni opened the pull request:
# tsParticles 1.30.2

## Bug Fixes

- Fixed gravity on retina displays and an issue with inverse gravity
- Fixed image drawer when refreshing the container

matteobruni committed:
fix: fixed start value for size and opacity, previous fix wasn't good enough

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
build: refactoring in the Updater.ts file, smaller output

matteobruni committed:
build: fixed solid.js readme and website updated with all frameworks

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.41.0 to 5.41.1

Bumps [webpack](https://github.com/webpack/webpack) from 5.41.0 to 5.41.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.41.0...v5.41.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump core-js from 3.15.1 to 3.15.2

Bumps [core-js](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js) from 3.15.1 to 3.15.2.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/commits/v3.15.2/packages/core-js)

---
updated-dependencies:
- dependency-name: core-js
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1727 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.41.1

build(deps-dev): bump webpack from 5.41.0 to 5.41.1

matteobruni committed:
Merge pull request #1728 from matteobruni/dependabot/npm_and_yarn/dev/core-js-3.15.2

build(deps): bump core-js from 3.15.1 to 3.15.2

matteobruni committed:
fix: fixed gravity on retina displays and an issue with inverse gravity

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
fix: fixed image drawer when refreshing the container

ghost commented:
## DeepCode's analysis on [#d75d31](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ba082b09892f73d3a4e5b8de0a9328116fe09683/matteobruni/tsparticles/d75d316653bb8fbc43f514303ee7629ef664c37e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) found:

- :warning: **1** warning :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/d75d316653bb8fbc43f514303ee7629ef664c37e/demo/vanilla/app.js#L41">app.js:41</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/d75d316653bb8fbc43f514303ee7629ef664c37e/demo/vanilla/app.js#L45">app.js:45</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/d75d316653bb8fbc43f514303ee7629ef664c37e/demo/vanilla/app.js#L49">app.js:49</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ba082b09892f73d3a4e5b8de0a9328116fe09683/matteobruni/tsparticles/d75d316653bb8fbc43f514303ee7629ef664c37e/pr/_/%2Fdemo%2Fvanilla%2Fapp.js/javascript%2Fdc_interfile_project%2FNoRateLimitingForExpensiveWebOperation/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ba082b09892f73d3a4e5b8de0a9328116fe09683/matteobruni/tsparticles/d75d316653bb8fbc43f514303ee7629ef664c37e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

matteobruni committed:
build: fixed demo imports package.json files

matteobruni committed:
build: fixed version scripts

matteobruni committed:
chore(release): published new version

 - ng-particles@2.13.2
 - inferno-particles@1.13.2
 - jquery-particles@1.30.2
 - preact-particles@1.30.2
 - react-tsparticles@1.30.2
 - riot-particles@1.4.2
 - solid-particles@1.13.2
 - svelte-particles@1.13.2
 - particles.vue@2.13.2
 - particles.vue3@1.13.2
 - web-particles@1.3.2
 - tsparticles-editor@1.30.2
 - tsparticles@1.30.2
 - ng-particles-demo@1.15.2
 - editor-demo@1.15.2
 - imports-ts-latest@1.1.2
 - imports-ts34@1.1.2
 - imports-ts38@1.1.2
 - imports-ts40@1.1.2
 - inferno-particles-demo@1.13.2
 - jquery-particles-demo@1.15.2
 - preact-particles-demo@1.15.2
 - react-tsparticles-demo@1.15.2
 - riot-particles-demo@1.4.2
 - solid-particles-demo@1.13.2
 - svelte-demo@1.13.2
 - tsparticles-demo@1.15.2
 - particles.vue-demo@1.15.2
 - particles.vue3-demo@1.13.2
 - web-particles-demo@1.15.2
 - tsparticles-preset-big-circles@1.15.2
 - tsparticles-preset-bubbles@1.15.2
 - tsparticles-preset-confetti@1.15.2
 - tsparticles-preset-fire@1.15.2
 - tsparticles-preset-firefly@1.15.2
 - tsparticles-preset-fireworks@1.15.2
 - tsparticles-preset-fountain@1.15.2
 - tsparticles-preset-links@1.15.2
 - tsparticles-preset-sea-anemone@1.15.2
 - tsparticles-preset-snow@1.15.2
 - tsparticles-preset-stars@1.15.2
 - tsparticles-shape-bubble@1.15.2
 - tsparticles-shape-heart@1.15.2
 - tsparticles-shape-multiline-text@1.15.2
 - tsparticles-shape-rounded-rect@1.15.2
 - tsparticles-shape-spiral@1.15.2
 - cra-template-particles-typescript@1.13.2
 - cra-template-particles@1.13.2
 - tsparticles-website@1.30.2

matteobruni committed:
build: fixed version scripts
